src/reporting/file-issue-for-review.js

```diff
@@ -0,0 +1,146 @@
+    trCrawlResultsPath = path.join(trCrawlResultsPath, 'index.json');
+  }
+  (async function() {
+    console.log(`Opening crawl results ${edCrawlResultsPath} and ${trCrawlResultsPath}…`)
+    const crawl = await loadCrawlResults(edCrawlResultsPath, trCrawlResultsPath);
+    console.log("- done");
+    console.log("Running broken link detection…");
+    const results = await studyBackrefs(crawl.ed, crawl.tr);
+    console.log("- done");
+    const currentBranch = execSync('git branch --show-current', { encoding: 'utf8' }).trim();
+    let counter = 0;
+    const needsPush = {};
+    await Promise.all(Object.keys(results).map(async uri => {
+      const specResult = results[uri];
+      console.log(`Compiling report for ${specResult.title}…`);
+      const brokenLinks = specResult.brokenLinks || [];
+      if (brokenLinks.length) {
+
+	if (counter > MAX_PR_BY_RUNS) return;
+	const issueMoniker = `${specResult.shortname}-brokenlinks`;
+	// is there already a file with that moniker?
+	const issueFilename = path.join('issues/', issueMoniker + '.md');
+	try {
+	  if (!(await fs.stat(issueFilename)).isFile()) {
+	    console.error(`${issueFilename} already exists but is not a file`);
+	  } else {
+	    console.log(`${issueFilename} already exists, bailing`);
+	  }
+	  return;
+	} catch (err) {
+	  // Intentionally blank
+	}
+	// if not, is there a pull request that uses that moniker as a branch
+	const {data: pullrequests} = (await octokit.rest.pulls.list({
+	  owner: repoOwner,
+	  repo: repoName,
+	  head: `${repoOwner}:${issueMoniker}`
+	}));
+	if (pullrequests.length > 0) {
+	  console.log(`A pull request from branch ${issueMoniker} already exists, bailing`);
+	  return;
+	}
+	// if not, we create the file, add it in a branch
+	// and submit it as a pull request to the repo
+	const issueReport = `
+Repo: ${specResult.repo}
+Tracked: N/A
+Issue title: Broken references in ${specResult.title}
+
+---
+
+While crawling [${specResult.title}](${specResult.crawled}), the following links to other specifications were detected as pointing to non-existing anchors, which should be fixed:
+${brokenLinks.map(link => `* [ ] ${link}`).join("\n")}
+
+<sub>This issue was detected and reported semi-automatically by [strudy](https://github.com/w3c/strudy/) based on data collected in [webref](https://github.com/w3c/webref/).</sub>`;
+	await fs.writeFile(issueFilename, issueReport, 'utf-8');
+	try {
+	  console.log(`Commiting issue report as ${issueFilename} in branch ${issueMoniker}…`);
+	  execSync(`git checkout -b ${issueMoniker}`);
+	  execSync(`git add ${issueFilename}`);
+	  execSync(`git commit -m "File report on broken links found in ${specResult.title}"`);
+	  needsPush[issueMoniker] = {title: `Broken references in ${specResult.title}`, report: issueReport, repo: specResult.repo, specTitle: specResult.title, uri: specResult.crawled, repo: specResult.repo};
+	  counter++;
+	  console.log("- done");
+	  execSync(`git checkout ${currentBranch}`);
```

Comment:
The code above should essentially be the same regardless of the report that we'll want to create. It currently contains bits specific to "broken links". No urgency, factoring out this code can certainly wait until we add another type of report!

Reply:
my thoughts exactly :)